File: aoc/aoc24/d1/part2.py
import sys
from collections import Counter
a, b = [], []
for line in sys.stdin:
    x, y = map(int, line.split())
    a.append(x)
    b.append(y)

c = Counter(b)
s = 0
for x in a:
    s += c.get(x, 0)*x
print(s)

File: aoc/aoc24/d1/part2.rs
use std::collections::HashMap;

fn main() {
    let mut v1 = Vec::new();
    let mut counter = HashMap::new();
    let stdin = std::io::stdin();
    for line in stdin.lines().map(|x| x.unwrap()) {
        let mut foo = line.split_ascii_whitespace();
        let a: i32 = foo.next().unwrap().parse().unwrap();
        let b: i32 = foo.next().unwrap().parse().unwrap();
        v1.push(a);
        *counter.entry(b).or_insert(0) += 1;
    }
    let s: i32 = v1.iter().map(|x| *x * *counter.entry(*x).or_default()).sum();
    println!("{}", s)
}
